src/main/java/org/jenkins/tools/test/model/hook/PluginCompatTesterHooks.java

```diff
@@ -106,25 +114,18 @@ public Map<String, Object> runBeforeExecution(Map<String, Object> elements) {
      * @param elements relevant information to hooks at various stages.
      */
     private Map<String, Object> runHooks(String stage, Map<String, Object> elements)
-            throws RuntimeException {
+            throws PluginCompatibilityTesterException {
         Queue<PluginCompatTesterHook> beforeHooks = getHooksFromStage(stage, elements);
 
         // Loop through hooks in a series run in no particular order
         // Modifications build on each other, pertinent checks should be handled in the hook
         for (PluginCompatTesterHook hook : beforeHooks) {
-            try {
-                if (!excludeHooks.contains(hook.getClass().getName()) && hook.check(elements)) {
-                    LOGGER.log(Level.INFO, "Running hook: {0}", hook.getClass().getName());
-                    elements = hook.action(elements);
-                    hook.validate(elements);
-                } else {
-                    LOGGER.log(Level.FINE, "Skipping hook: {0}", hook.getClass().getName());
-                }
-            } catch (RuntimeException re) {
-                // this type of exception should stop processing the plugins. Throw it up the chain
-                throw re;
-            } catch (Exception ex) {
-                LOGGER.log(Level.WARNING, "Failed to run hook; continuing", ex);
+            if (!excludeHooks.contains(hook.getClass().getName()) && hook.check(elements)) {
+                LOGGER.log(Level.INFO, "Running hook: {0}", hook.getClass().getName());
+                elements = hook.action(elements);
+                hook.validate(elements);
+            } else {
+                LOGGER.log(Level.FINE, "Skipping hook: {0}", hook.getClass().getName());
             }
```

Comment:
Important: we no longer swallow arbitrary checked exceptions in hooks. Hook exceptions now propagate upward like anything else.